postgres/datadog_checks/postgres/statements.py

```diff
@@ -186,6 +199,40 @@ def _get_pg_stat_statements_columns(self):
                 self._stat_column_cache = col_names
                 return col_names
 
+    def _check_called_queries(self):
+        pgss_view_without_query_text = self._config.pg_stat_statements_view
+        if pgss_view_without_query_text == "pg_stat_statements":
+            # Passing false for the showtext argument leads to a huge performance increase. This
+            # allows the engine to avoid retrieving the potentially large amount of text data.
+            # The query count query does not depend on the statement text, so it's safe for this use case.
+            # For more info: https://www.postgresql.org/docs/current/pgstatstatements.html#PGSTATSTATEMENTS-FUNCS
+            pgss_view_without_query_text = "pg_stat_statements(false)" 
+
+        with self._check._get_main_db() as conn:
+            with conn.cursor(cursor_factory=CommenterCursor) as cursor:
+                called_queryids = []
+                query = QUERYID_TO_CALLS_QUERY.format(pg_stat_statements_view=pgss_view_without_query_text)
+                rows = self._execute_query(cursor, query, params=(self._config.dbname,))
+                
+                for row in rows:
+                    queryid = row[0]
+                    if queryid is None:
+                        continue
+
+                    calls = row[1]
+                    calls_changed = self._query_calls_cache.set_calls(queryid, calls)
+                    if calls_changed:
+                        called_queryids.append(queryid)
+                self._query_calls_cache.end_query_call_snapshot()
```

Comment:
Small suggestion to make `called_queryids` be a _query_calls_cache attribute. We can do something like
```suggestion
                for row in rows:
                    self._query_calls_cache.set_calls(*row)
                self._query_calls_cache.end_query_call_snapshot()
```
For `QueryCallsCache`
```python
def __init__(self):
        self.cache = {}
        self.next_cache = {}
        self.called_queryids = set()

def end_query_call_snapshot(self):
        self.cache = self.next_cache
        self.next_cache = {}
        self.called_queryids.clear()

def set_calls(self, queryid, calls):
        if not queryid:
            return
        if queryid not in self.cache or calls - self.cache[queryid] != 0:
            self.called_queryids.add(queryid)
        self.next_cache[queryid] = calls
```

Reply:
🙇‍♂️